File: book-examples/02-kmeans-movie-review/mapper.R
#! /usr/bin/env Rscript

input <- file("stdin","r")

A<-c()
B<-c()

while(length(currentLine <- readLines(input,n=1,warn=FALSE))>0)
{
    fields <- unlist(strsplit(currentLine,","))
    a=as.double(fields[1])
    b=as.double(fields[2])
    A=c(A,a)
    B=c(B,b)
}
mydata <- data.frame(A,B)

scale01=function(x){
    ncol=dim(x)[2]
    nrow=dim(x)[1]
    new=matrix(0,nrow,ncol)
    for(i in 1:ncol)
    {max=max(x[,i])
        min=min(x[,i])
        for(j in 1:nrow)
        {new[j,i]=(x[j,i]-min)/(max-min)}
    }
    new
}

datanorm=scale01(mydata)
print(datanorm,stdout())
close(input)

File: book-examples/02-kmeans-movie-review/mapper.py
#! /usr/bin/env python
from itertools import combinations
import sys
import numpy as np
import pandas as pd

A=[]
B=[]

for line in sys.stdin:
    fields = line.strip().split(',')
    a = float(fields[0])
    b = float(fields[1])
    A.append(a)
    B.append(b)
    for i in range(1,len(A)):
	A[i] = (A[i] - min(A))/(max(A)-min(A))
    for j in range(1,len(B)):
	B[j] = (B[j] -min(B))/(max(B)-min(B))

A = np.array(A)
B = np.array(B)
data = np.column_stack((A , B))
print(data)
sys.stdin.close()
